### PR TITLE
Enable passing features flags to cargo when expanding macros

### LINKED ISSUE #189: [Feature] Feature Flags
Currently no option exists to enable feature flags in doc2readme. This means doc2readme can to create links for functions, which are not available with the default features.

For example in https://github.com/LuckyTurtleDev/more-wallpapers/pull/7#pullrequestreview-1192074089 the link for `set_random_wallpapers_from_vec()` cannot be created. Instead, doc2readme create a link to a non existing `set_random_wallpapers_from_vec()` file.

### PR DESCRIPTION
**Pull request by LuckyTurtleDev**
Fixes #189

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -115,6 +115,18 @@ struct Args {
 	#[clap(long)]
 	expand_macros: bool,
 
+	/// Space or comma separated list of features to activate
```
Comment (msrd0) [inline src/main.rs:118]: can features really be separated by comma?

File: src/main.rs [hunk 2]
```diff
@@ -115,6 +115,18 @@ struct Args {
...
+	#[clap(long)]
```
Comment (msrd0) [inline src/main.rs:119]: Please also add the short `-F` flag which cargo also supports.

File: src/main.rs [hunk 3]
```diff
@@ -155,6 +167,23 @@ fn main() -> ExitCode {
 		_ => Args::parse()
 	};
 
+	//check input
+	if !args.expand_macros {
+		let mut diag = Diagnostic::new(String::new(), String::new());
+		if args.features.is_some() {
+			diag.warn("--features option has no effect without the --expand-marcos flag")
```
Comment (msrd0) [inline src/main.rs:174]:
```suggestion
			diag.warn("--features option has no effect without the --expand-macros flag")
```

File: src/main.rs [hunk 4]
```diff
@@ -155,6 +167,23 @@ fn main() -> ExitCode {
...
+		}
+		if args.no_default_features {
+			diag.warn("--no_default_features flag has no effect without the --expand-marcos flag")
```
Comment (msrd0) [inline src/main.rs:177]:
```suggestion
			diag.warn("--no-default-features flag has no effect without the --expand-macros flag")
```

File: src/main.rs [hunk 5]
```diff
@@ -155,6 +167,23 @@ fn main() -> ExitCode {
...
+		}
+		if args.all_features {
+			diag.warn(
+				"--all-features option has no effect without the --expand-marcos flag"
```
Comment (msrd0) [inline src/main.rs:181]:
```suggestion
				"--all-features flag has no effect without the --expand-macros flag"
```

File: tests/pass/fn-link/lib.rs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+//! blah blah
+//! test [`foo()`] test
+//! blah blah
+
+/// do nothing
+pub fn foo(){}
```
Comment (msrd0) [inline tests/pass/fn-link/lib.rs:6]: Please add missing newline at the end of the file

File: tests/tests.rs [hunk 7]
```diff
@@ -6,12 +6,28 @@ use cargo_doc2readme::{output, read_input, verify};
...
 	fs::{self, File},
+	io,
 	panic::catch_unwind,
 	path::{Path, PathBuf}
 };
 
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(deny_unknown_fields)]
+struct TestConfig {
+	/// test need required nightly rust to run
```
Comment (msrd0) [inline tests/tests.rs:20]:
```suggestion
	/// test requires nightly rust to run
```

**Review by msrd0 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (LuckyTurtleDev) [reply src/main.rs:118]: no, I had simple copy the message from cargo and forget about it

Comment (msrd0) [reply src/main.rs]: well if cargo supports it than that's fine I guess

Comment (msrd0) [reply src/main.rs]: Actually, if cargo says it's `Space or comma separated list of features to activate` then we should keep it that way. We don't parse the features anyways, just pass them through to cargo

File: tests/tests.rs [hunk 8]
```diff
@@ -168,12 +192,37 @@ where
...
+					toml::from_str(&test_config).unwrap()
+				} else {
+					TestConfig::default()
+				};
+
+				if test_config.nightly && !rustversion::cfg!(nightly) {
+					continue;
+				}
+
+				if !rustversion::cfg!(nightly) {}
```
Comment (msrd0) [inline tests/tests.rs:218]: What does this `if` do?

Comment (LuckyTurtleDev) [reply tests/tests.rs:218]: nothing. I wonder why clippy odes not warn about it.

**Review by msrd0 — APPROVED** [review]
Thanks

**Pull request merged**